preprocess/sct-preprocess_data.sh

```diff
@@ -0,0 +1,461 @@
+    fi
+  fi
+}
+
+label_if_does_not_exist(){
+  ###
+  #  This function checks if a manual labels exists, then:
+  #    - If it does, copy it locally and use them to initialize vertebral labeling
+  #    - If it doesn't, perform automatic vertebral labeling
+  ###
+  local file="$1"
+  local file_seg="$2"
+  # Update global variable with segmentation file name
+  FILELABEL="${file}_labels"
+  FILELABELMANUAL="${PATH_DATA}/derivatives/labels/${SUBJECT}/anat/${FILELABEL}-manual.nii.gz"
+  echo "Looking for manual label: $FILELABELMANUAL"
+  if [[ -e $FILELABELMANUAL ]]; then
+    echo "Found! Using manual labels."
+    rsync -avzh $FILELABELMANUAL ${FILELABEL}.nii.gz
+  else
+    echo "Not found. Proceeding with automatic labeling."
+    # Generate vertebral labeling
+    sct_label_vertebrae -i ${file}.nii.gz -s ${file_seg}.nii.gz -c t2 -qc ${PATH_QC} -qc-subject ${SUBJECT}
+    # Create labels in the cord at C3 and C5 mid-vertebral levels
+    sct_label_utils -i ${file_seg}_labeled.nii.gz -vert-body 3,5 -o ${FILELABEL}.nii.gz -qc ${PATH_QC} -qc-subject ${SUBJECT}
+  fi
+}
+
+# Retrieve input params and other params
+SUBJECT=$1
+
+# get starting time:
+start=`date +%s`
+
+
+# SCRIPT STARTS HERE
+# ==============================================================================
+# Display useful info for the log, such as SCT version, RAM and CPU cores available
+sct_check_dependencies -short
+
+# Go to folder where data will be copied and processed
+cd $PATH_DATA_PROCESSED
+
+# Copy BIDS-required files to processed data folder (e.g. list of participants)
+if [[ ! -f "participants.tsv" ]]; then
+  rsync -avzh $PATH_DATA/participants.tsv .
+fi
+# Copy list of participants in results folder 
+if [[ ! -f "participants.json" ]]; then
+  rsync -avzh $PATH_DATA/participants.json .
+fi
+if [[ ! -f "dataset_description.json" ]]; then
+  rsync -avzh $PATH_DATA/dataset_description.json .
+fi
+#if [[ ! -f "README" ]]; then
+#  rsync -avzh $PATH_DATA/README .
+#fi
+
+# Copy source images
+# Note: we use '/./' in order to include the sub-folder 'ses-0X'
+rsync -Ravzh $PATH_DATA/./$SUBJECT .
+
+# # Copy segmentation ground truths (GT)
+# mkdir -p derivatives/labels
+# rsync -Ravzh $PATH_DATA/derivatives/labels/./$SUBJECT derivatives/labels/.
+
+# Go to subject folder for source images
+cd ${SUBJECT}/anat
+
+# Define variables
+# We do a substitution '/' --> '_' in case there is a subfolder 'ses-0X/'
+file="${SUBJECT//[\/]/_}"
+
+# -------------------------------------------------------------------------
+# T2w
+# -------------------------------------------------------------------------
+# Add suffix corresponding to contrast
+file_t2w=${file}_T2w
+# Check if T2w image exists
+if [[ -f ${file_t2w}.nii.gz ]];then
+    # Make sure the image metadata is a valid JSON object
+    if [[ ! -s ${file_t2w}.json ]]; then
+      echo "{}" >> ${file_t2w}.json
+    fi
+
+    # Rename raw file
+    mv ${file_t2w}.nii.gz ${file_t2w}_raw.nii.gz
```

Comment:
i'm not sure I get the logic of renaming with '_raw' suffix, if at the end the file is called ${file_t2w} (https://github.com/ivadomed/canproco/pull/5/files#diff-954472fe9469d17aa67299797673e161b522aa01949f615e1e0df9f2bc8f5d0dR203)

The idea is that the preprocessed files are output into a *new* BIDS dataset (eg called canproco-processed), without all the temporary junk, and this BIDS dataset will be input into ivadomed. See for example: https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/blob/1a43ed4a0ea72f9d872aa80838cb0bde6a8db3d5/process_data.sh#L404-L477

Reply:
That being said, maybe your approach is better and we could discuss the pros/cons at our next meeting.

Reply:
The idea of the `mv` command is to backup the original input file as a `_raw.nii.gz`. Then, after preprocessing steps (e.g., reorientation and resampling), a symbolic link is created, and segmentation is done to create a BIDS-compliant segmentation (i.e., without temporary suffixes such as `_RPI_r`). See, for example, lines [L195-L208](https://github.com/ivadomed/canproco/blob/1663095f82349fb18852ac3e9fddd9494da85056/preprocess/sct-preprocess_data.sh#L195-L208).
However, the PR is still a draft; I will be happy to discuss that at the meeting!

Reply:
> The idea of the mv command is to backup the original input file

My point was that a backup is unnecessary because we always have access to the source data (ie: stored at `data:canproco`). The purpose of this preprocessing script is to create a new BIDS-compliant dataset (eg: called `canproco-processed`) with _only_ the files needed for model training and data analysis.

Reply:
Thank you for the clarification, now I get it! I will update the code.